internal/resolution/variablesources/bundles_and_dependencies.go

```diff
@@ -63,25 +63,25 @@ func (b *BundlesAndDepsVariableSource) GetVariables(ctx context.Context) ([]depp
 		var head *catalogmetadata.Bundle
 		head, bundleQueue = bundleQueue[0], bundleQueue[1:]
 
-		for _, id := range olmvariables.BundleToBundleVariableIDs(head) {
-			// ignore bundles that have already been processed
-			if _, ok := visited[id]; ok {
-				continue
-			}
-			visited[id] = struct{}{}
-
-			// get bundle dependencies
-			dependencies, err := b.filterBundleDependencies(allBundles, head)
-			if err != nil {
-				return nil, fmt.Errorf("could not determine dependencies for bundle with id '%s': %w", id, err)
-			}
+		id := olmvariables.BundleVariableID(head)
 
-			// add bundle dependencies to queue for processing
-			bundleQueue = append(bundleQueue, dependencies...)
+		// ignore bundles that have already been processed
+		if _, ok := visited[id]; ok {
+			continue
+		}
+		visited[id] = struct{}{}
```

Comment:
Can we use `sets.New[deppy.Identifier]()` for clearer semantics on `visited` and `added`?

Reply:
AFAIK Go doesn't have sets in standard library and as a result using maps for sets is pretty common.

But we can do that. However I do not see any sets package in our codebase. What package is it?

Reply:
Anyway I think it should not be in this PR. Could you please create an issue or PR for this suggesting a pacakge to use?

Reply:
The k8s API machinery sets package is the one - we use it in basically every other package. Feel free to create the issue if you don't think you can do it here.

Reply:
I'm a fan of small PRs serve one porpuse. Don't want to mix different efforts into this PR.

Created #454 for this.